decompiler/frontend/binaryninja/frontend.py

```diff
@@ -24,6 +24,7 @@ class FunctionObject:
     def __init__(self, function: Function):
         self._function = function
         self._lifter = BinaryninjaLifter()
+        self._name = self._lifter.lift(self._function.symbol).name
```

Comment:
I'm afraid this could cause problems when the user actually renamed the function. Could you double check?